src/inmanta/data/__init__.py

```diff
@@ -5578,6 +5622,34 @@ async def get_increment(
         # todo in this version
         work: list[abc.Mapping[str, object]] = [r for r in resources if r["status"] not in UNDEPLOYABLE_NAMES]
 
+        # start with outstanding events
+        id_to_resource = {r["resource_id"]: r for r in resources}
+        next: list[abc.Mapping[str, object]] = []
+        for resource in work:
```

Comment:
Wouldn't it be more performant to do this in postgres? If you want to get it merged today that's fine, we can address this later.